platform/viewer/src/appExtensions/MeasurementsPanel/init.js

```diff
@@ -46,17 +46,95 @@ export default function init({
     'labelmapModified'
   );
 
+  const _getDefaultPosition = event => ({
+    x: (event && event.currentPoints.client.x) || 0,
+    y: (event && event.currentPoints.client.y) || 0,
+  });
+
+  const _updateLabellingHandler = (labellingData, measurementData) => {
+    const { location, description, response } = labellingData;
+
+    if (location) {
+      measurementData.location = location;
+    }
+
+    measurementData.description = description || '';
+
+    if (response) {
+      measurementData.response = response;
+    }
+
+    commandsManager.runCommand(
+      'updateTableWithNewMeasurementData',
+      measurementData
+    );
+  };
+
+  const showLabellingDialog = (props, contentProps, measurementData) => {
+    if (UIDialogService) {
+      UIDialogService.create({
+        id: 'labelling',
+        isDraggable: false,
+        showOverlay: true,
+        content: LabellingFlow,
+        contentProps: {
+          measurementData,
+          labellingDoneCallback: () =>
+            UIDialogService.dismiss({ id: 'labelling' }),
+          updateLabelling: labellingData =>
+            _updateLabellingHandler(labellingData, measurementData),
+          ...contentProps,
+        },
+        ...props,
+      });
+    }
+  };
+
   const onRightClick = event => {
-    if (UIContextMenuService) {
-      UIContextMenuService.show({ event: event.detail });
+    if (UIDialogService) {
+      UIDialogService.dismiss({ id: 'context-menu' });
```

Comment:
To reduce nesting, we could return early:

```js
if(!UIDialogService){
  console.warn('Unable to show tool context menu; no UI Dialog Service available.');
  return;
}
```

Reply:
Alright, its more informative too. Done! We can find a way to proxy all service calls to avoid this type of check, lets keep this in mind as well. Thanks.